lock-api-objects/src/main/java/com/palantir/lock/watch/LockEvent.java

```diff
@@ -16,30 +16,40 @@
 
 package com.palantir.lock.watch;
 
-import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.annotation.JsonTypeName;
-import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
-import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
 import com.palantir.lock.LockDescriptor;
 import com.palantir.lock.v2.LockToken;
+import com.palantir.lock.watch.serialization.WireLockRequestMetadata;
 import com.palantir.logsafe.Unsafe;
+import com.palantir.util.IndexEncodingUtils;
+import com.palantir.util.IndexEncodingUtils.ChecksumType;
+import com.palantir.util.IndexEncodingUtils.IndexEncodingResult;
+import com.palantir.util.IndexEncodingUtils.KeyListChecksum;
+import com.palantir.util.Pair;
+import java.util.List;
+import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
+import java.util.function.Function;
 import org.immutables.value.Value;
 
 @Unsafe
 @Value.Immutable
-@JsonSerialize(as = ImmutableLockEvent.class)
-@JsonDeserialize(as = ImmutableLockEvent.class)
 @JsonTypeName(LockEvent.TYPE)
 public abstract class LockEvent implements LockWatchEvent {
     static final String TYPE = "lock";
+    private static final ChecksumType DEFAULT_CHECKSUM_TYPE = ChecksumType.CRC32_OF_DETERMINISTIC_HASHCODE;
 
     public abstract Set<LockDescriptor> lockDescriptors();
 
     public abstract LockToken lockToken();
 
-    @JsonIgnore
     public abstract Optional<LockRequestMetadata> metadata();
```

Comment:
This does not enable the serialization of metadata since we now have a custom `@JsonValue` method